File: socket_client_ws.js
const WebSocket = require("ws");

// Function to handle incoming messages from the WebSocket server
function onMessage(ws, message) {
  console.log(`Received message: ${message}`);
}

// Function to handle WebSocket errors
function onError(ws, error) {
  console.error(`Error: ${error}`);
}

// Function to handle WebSocket closure
function onClose(ws, code, reason) {
  console.log("WebSocket closed");
}

// Function to handle WebSocket connection open event
function onOpen(ws) {
  console.log("WebSocket connection opened");
  const testMessage = "Test message from JavaScript";
  console.log(`Sending message: ${testMessage}`);
  ws.send(testMessage);
  startPing(ws);
}

// Function to send ping message every 5 seconds
function startPing(ws) {
  function ping() {
    if (ws.readyState === WebSocket.OPEN) {
      console.log("Sending ping...");
      ws.send("ping-socket-javascript");
    } else {
      console.log("WebSocket is not connected, skipping ping.");
    }
    setTimeout(ping, 5000);
  }

  ping();
}

// Main function to establish WebSocket connection
function main() {
  const brokerUrl =
    "ws://localhost:56567/websocket?sessionId=javascript-session"; // Notice: ws:// instead of wss://
  console.log(`Connecting to WebSocket at ${brokerUrl}...`);

  // Create the WebSocket client without HTTPS agent
  const ws = new WebSocket(brokerUrl);

  // Setup event listeners
  ws.on("open", () => onOpen(ws));
  ws.on("message", (message) => onMessage(ws, message));
  ws.on("error", (error) => onError(ws, error));
  ws.on("close", (code, reason) => onClose(ws, code, reason));
}

main();

File: socket_client_ws.py
import websocket
import time
import threading

def on_message(ws, message):
    print(f"Received message: {message}")

def on_error(ws, error):
    print(f"Error: {error}")

def on_close(ws, close_status_code, close_msg):
    print("WebSocket closed")

def on_open(ws):
    print("WebSocket connection opened")
    test_message = 'Test message from Python'
    print(f"Sending message: {test_message}")
    ws.send(test_message)
    start_ping(ws)

def start_ping(ws):
    def ping():
        if ws and ws.sock and ws.sock.connected:
            try:
                print("Sending ping...")
                ws.send('ping-socket-python-ip-130')
            except Exception as e:
                print(f"Error sending ping: {e}")
        else:
            print("WebSocket is not connected, skipping ping.")
        
        threading.Timer(5, ping).start()
    
    ping()

def main():
    # broker_url = 'ws://192.168.1.130:60192/websocket?sessionId=python-session'  # Changed to ws://
    broker_url = 'ws://localhost:60288/websocket?sessionId=python-session'  # Changed to ws://
    print(f"Connecting to WebSocket at {broker_url}...")

    ws = websocket.WebSocketApp(
        broker_url,
        on_message=on_message,
        on_error=on_error,
        on_close=on_close,
        on_open=on_open
    )

    # No SSL options for plain ws://
    ws.run_forever()

if __name__ == '__main__':
    main()
